akka-docs-dev/rst/scala/code/docs/stream/GraphStageDocSpec.scala

```diff
@@ -83,4 +87,423 @@ class GraphStageDocSpec extends AkkaSpec {
     Await.result(result2, 3.seconds) should ===(5050)
   }
 
+  //#one-to-one
+  class Map[A, B](f: A => B) extends GraphStage[FlowShape[A, B]] {
+
+    val in = Inlet[A]("Map.in")
+    val out = Outlet[B]("Map.out")
+
+    override def shape = FlowShape.of(in, out)
+
+    override def createLogic(attr: Attributes): GraphStageLogic =
+      new GraphStageLogic(shape) {
+        setHandler(in, new InHandler {
+          override def onPush(): Unit = {
+            val a = grab(in)
+            val b = f(a)
```

Comment:
Maybe I should just inline it instead? Did this because I thought `push(out, f(grab(in)))` was to dense, but on second thought I like that better.